subsys/net/lib/sockets/sockets_inet.c

```diff
@@ -1973,6 +1973,21 @@ int zsock_getsockopt_ctx(struct net_context *ctx, int level, int optname,
 			}
 
 			break;
+		case IP_MULTICAST_LOOP:
+			if (IS_ENABLED(CONFIG_NET_IPV4)) {
+				if (*optlen != sizeof(int)) {
+					errno = EINVAL;
+					return -1;
+				}
+		
+				*(int *)optval = ctx->ipv4_mcast_loop;
+				*optlen = sizeof(int);
+				
+				return 0;
+			}
+				
+			errno = ENOPROTOOPT;
+			return -1;
```

Comment:
The `ipv4_mcast_loop` member is compiled out if `CONFIG_NET_IPV4` isn't enabled.

```suggestion
#if defined(CONFIG_NET_IPV4)
		case IP_MULTICAST_LOOP:
			if (*optlen != sizeof(int)) {
				errno = EINVAL;
				return -1;
			}
		
			*(int *)optval = ctx->ipv4_mcast_loop;
			*optlen = sizeof(int);
				
			return 0;
#endif
```